### PR TITLE
Switch to the assets plugin

### PR DESCRIPTION
**Pull request by abarth**
This patch removes our dependency on asset_bundle.mojom.

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie @chinmaygarde @jason-simmons

File: packages/flutter/lib/src/services/image_provider.dart [hunk 1]
```diff
@@ -209,26 +208,62 @@ abstract class ImageProvider<T> {
...
 ///
-/// This factors out the common logic of many [ImageProvider] classes,
-/// simplifying what subclasses must implement to just three small methods:
+/// This is used to identify the precise resource in the [imageCache].
+class AssetBundleImageKey {
+  /// Creates the key for an [AssetImage] or [AssetBundleImageProvider].
+  ///
+  /// The arguments must not be null.
+  const AssetBundleImageKey({
+  @required this.bundle,
```
Comment (Hixie) [inline packages/flutter/lib/src/services/image_provider.dart]: indentation

File: packages/flutter/lib/src/services/image_provider.dart [hunk 2]
```diff
@@ -209,26 +208,62 @@ abstract class ImageProvider<T> {
   String toString() => '$runtimeType()';
 }
 
-/// A subclass of [ImageProvider] that knows how to invoke
-/// [decodeImageFromDataPipe].
+/// Key for the image obtained by an [AssetImage] or [AssetBundleImageProvider].
```
Comment (Hixie) [inline packages/flutter/lib/src/services/image_provider.dart]: I think you mean `[AssetImage] or [ExactAssetImage]`.

**Comment by Hixie** [comment]
LGTM

**Pull request merged**